### PR TITLE
Desktope2e remixai

### PR DESCRIPTION
**Pull request by STetsing**
# Introducing local AI on Remix desktop IDE
This PR brings the already known features to the desktop environment while enabling the local AI inference on user machine.
The user can still opt for remote inference similar as in the browser

## Currently supported targets platforms
- ARM MacOS
- Windows
- ARM linux 
see [InferenceServer releases](https://github.com/remix-project-org/remix_ai_tools/releases/tag/v0.0.2)

### CODE REVIEW

File: apps/remix-ide/src/app/plugins/remixAIPlugin.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    // user machine dont use ressource for remote inferencing
+  }
+
+  onActivation(): void {
+    if (this.isOnDesktop) {
+      console.log('Activating RemixAIPlugin on desktop')
+    } else {
+      console.log('Activating RemixAIPlugin on browser')
+      this.initialize()
```
Comment (STetsing) [inline apps/remix-ide/src/app/plugins/remixAIPlugin.tsx:46]: @yann300 this automatically enable the remix AI on browser on activation

File: apps/remix-ide/src/app.js [hunk 2]
```diff
@@ -388,11 +400,22 @@ class AppComponent {
...
       const ripgrep = new ripgrepPlugin()
       this.engine.register([ripgrep])
+      const appUpdater = new appUpdaterPlugin()
+      this.engine.register([appUpdater])
+      const remixAIDesktop = new remixAIDesktopPlugin()
+      this.engine.register([remixAIDesktop])
     }
+    
+    const remixAI = new RemixAIPlugin(isElectron())
+    this.engine.register([remixAI])
```
Comment (STetsing) [inline apps/remix-ide/src/app.js:410]: @yann300 activate this plugin on startup, since it is not view anymore and requires user activate

File: apps/remix-ide/src/app/plugins/remixAIPlugin.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  icon: 'assets/img/remix-logo-blue.png',
+  description: 'RemixAI provides AI services to Remix IDE.',
+  kind: '',
+  location: 'sidePanel',
+  documentation: 'https://remix-ide.readthedocs.io/en/latest/remixai.html',
+  version: packageJson.version,
+  maintainedBy: 'Remix'
+}
+
+export class RemixAIPlugin extends ViewPlugin {
```
Comment (STetsing) [inline apps/remix-ide/src/app/plugins/remixAIPlugin.tsx:26]: @yann300 turn this to simple plugin

File: apps/remix-ide/src/app/tabs/search.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import React from 'react' // eslint-disable-line
 import { SearchTab } from '@remix-ui/search'
 const profile = {
   name: 'search',
-  displayName: 'Search in files',
+  displayName: 'Searchi in files',
```
Comment (yann300) [inline apps/remix-ide/src/app/tabs/search.tsx:7]: typo

File: apps/remixdesktop/src/lib/slither.ts [hunk 5]
```diff
@@ -0,0 +1,193 @@
+import { existsSync, readFileSync, readdirSync, unlinkSync } from 'fs'
```
Comment (yann300) [inline apps/remixdesktop/src/lib/slither.ts:1]: this diff shouldn't be part of the PR

File: libs/remix-ai-core/package-lock.json [hunk 6]
```diff
@@ -0,0 +1,5344 @@
+{
```
Comment (yann300) [inline libs/remix-ai-core/package-lock.json:1]: there shouldn't be a lock file here.

File: libs/remix-ai-core/yarn.lock [hunk 7]
```diff
@@ -0,0 +1,2186 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (yann300) [inline libs/remix-ai-core/yarn.lock:1]: same here, I don't think there should be lock filesin `libs`

File: package-lock.json
Comment (yann300) [inline package-lock.json:1]: no lock file

File: apps/remixdesktop/src/lib/slither.ts (hunk 5 above)
Comment (STetsing) [reply apps/remixdesktop/src/lib/slither.ts:1]: this comes from master. 
How to handle that? complete removal?

Comment (yann300) [reply apps/remixdesktop/src/lib/slither.ts:1]: is it? I don't see it in master

File: apps/remixdesktop/src/plugins/remixAIDektop.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+  documentation: 'https://remix-ide.readthedocs.io/en/latest/remixai.html',
+}
+
+export class RemixAIDesktopPlugin extends ElectronBasePlugin {
+  clients: RemixAIDesktopPluginClient[] = []
+  constructor() {
+    super(profile, clientProfile, RemixAIDesktopPluginClient)
+    this.methods = [...super.methods]
+
+    for (const client of this.clients) {
```
Comment (bunsenstraat) [inline apps/remixdesktop/src/plugins/remixAIDektop.ts:26]: normally clients get initiated when connecting remix IDE, and that will only happen after the constructor. in the BasePlugin you add methods that can talk to the clients, but on load that [] will be empty

File: apps/remixdesktop/src/lib/slither.ts (hunk 5 above)
Comment (bunsenstraat) [reply apps/remixdesktop/src/lib/slither.ts:1]: @STetsing remove the file

**Review by yann300 — APPROVED** [review]
